rfcs/rfc-bld-20230302-1-3p-development.md

```diff
@@ -0,0 +1,242 @@
+-------------------------
+The advantages to this approach:
+
+*   All scripts can be contained in the .github/workflow folder, which will follow along with any fork or branch.
+*   Github Actions is invoked within the scope of the repo, fork, or branch, which decentrializes its output
+*   Github Actions environments are containerized, so that a developer can reproduce builds locally
+*   Github packages in OCI format come with "free" vulnerability scanning, supply chain protection (containers can be signed), and traceability (the artifact hash can be correlated to other public packages)
+*   For public repos, this infrastructure is "free" of cost, with some nebulous limit but seems higher than the sum of our current packages. The only time costs come into play is for private repos, which already come with a bucket of Actions minutes and Packages storage space.
+
+$~$
+
+There's some cons however:
+*   Because the development package output is decentralized, we won't have a central "development" S3 bucket. 
+    - *Mitigation:* This means a contributor will have to ensure that they add the Github packages URL for their fork to the the list of CDN urls to be pulled via cmake for iterative development.
+*   Github Actions uses a containerized infrastructure, with limited specs - something on the order of [2 CPUs with 7 GB RAM](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) with 19~30GB of SSD storage. Large packages like QT and Python may take hours to build. 
+    - *Mitigation:* There may be ways to utilize Linux Foundation's hardware as "agents" (called self-hosted runners) for Github Actions if we need higher specs
+*   Support for versions of OS is on a ["window" of releases](https://github.com/actions/virtual-environments#available-environments). Some OS versions will fall out of support.
+    - *Mitigation:* Older supported platforms will be augmented by self-hosted runners
+*   We're at the mercy of Github for any support and if Github itself goes down, then users pulling 3p from an installed version of O3DE may get blocked. 
+    - *Mitigation:* We will still use our current Cloudfront/S3 CDN on [o3debinaries.org](http://o3debinaries.org) as a backup, but other CDNs can be used as a potential mirror. This is controlled by the CDN list in [https://github.com/o3de/o3de/blob/development/cmake/3rdPartyPackages.cmake#L32](https://github.com/o3de/o3de/blob/development/cmake/3rdPartyPackages.cmake#L32)
+
+Security and license Considerations
+-------------------------
+*   How to we prevent supply chain attacks?
+    - Github Action trigger abuse
+        - *Potential threat:* Threat actors DDoSing Github Actions or causing it to trigger automatically and merge without review
+        - *Mitigation:* Only maintainers will be able to trigger the GHA manually. We require a PR of the contribution before it is merged, and must be signed off by 2 people

```

Comment:
this specifically refers to the build step, I'm guessing? (only triggered by maintainers, only manually)

Reply:
Yup, lemme re-write that to make it more clear